app/.helper/package

```diff
@@ -43,38 +68,111 @@ get_package_channel() {
     fi
 }
 
-# get_latest_version fetches the latest numerical version of a package from the
-# repository
-get_latest_version() {
-    release_repo=$1
+# get_package_platform parses a package's target platform from its canonical
+# name. Returns an empty string when the package canonical name does not specify
+# the targeted platform.
+get_package_platform() {
+    # Unlike package name and channel, platform can contain dashes, and thus
+    # cannot be splitted with awk.
+
+    if [[ "$(kernel)" == "Darwin" ]]; then
+        echo "$1" | sed -nE 's/^[a-zA-Z0-9]+-[a-zA-Z0-9.]+-[a-zA-Z0-9]+-?(.*)$/\1/p'
```

Comment:
I think we can simplify the code by putting the `-nE` or `-nr` flags inside a variable instead?

```
SED_OPTS="-nr"
if [[ "$(kernel)" == "Darwin" ]]; then
  SED_OPTS="-nE"
fi

echo "$1" | sed $SED_OPTS 's/^[a-zA-Z0-9]+-[a-zA-Z0-9.]+-[a-zA-Z0-9]+-?(.*)$/\1/p'
```

Since the regex is quite complex, we can avoid changing it twice whenever we need to update it by having just the flags for `sed` configured by the `if` block instead.

Reply:
Good idea, I'll fix this in the next commit